components/movementsensor/gpsnmea/pmtkI2C.go

```diff
@@ -144,14 +136,14 @@ func (g *PmtkI2CNMEAMovementSensor) Start(ctx context.Context) error {
 				handle, err := g.bus.OpenHandle(g.addr)
 				if err != nil {
 					g.logger.Errorf("can't open gps i2c handle: %s", err)
-					g.setLastError(err)
+					g.err.Set(err)
 					return
 				}
 				buffer, err := handle.Read(ctx, 1024)
 				hErr := handle.Close()
 				if hErr != nil {
 					g.logger.Errorf("failed to close handle: %s", hErr)
-					g.setLastError(err)
+					g.err.Set(err)
```

Comment:
Something in here is muddled: we will only store `err` if `hErr` is non-nil. and `hErr` is never stored for later, only logged. I kinda want to fix that in a separate PR; this one is already too large.

Reply:
perhaps a `multierror.Combine` for both errors would be sufficient here?

Reply:
two things - the reading code right below is important to keep, it's the reason we clear the error right now actually, since it tires it's best to retry reading and parsing until it succeeds in getting a message. So whatever function implements the `defer` would have to be rewritten to deal with only Open, Read, Close i2cHandles and return an error.